### PR TITLE
precommit checks && repo hygiene

### PR DESCRIPTION
**Pull request by jamiezieziula**
## Description
Introduce additional precommit checks & general cleanup of the repo

Resolves https://github.com/PrefectHQ/prefect-recipes/issues/171

### CODE REVIEW

File: devops/infrastructure-as-code/azure/prefect-agent-on-aks/prefect.yaml [hunk 1]
```diff
@@ -14,20 +15,30 @@ spec:
...
-      - name: agent
-        image: prefecthq/prefect:2.0b9-python3.9
-        command: ["prefect", "agent", "start", "kubernetes"]
-        imagePullPolicy: "IfNotPresent"
-        env:
-          - name: PREFECT_API_URL
-            value: http://orion:4200/api
-
+        - name: api
+          image: prefecthq/prefect:2.0b9-python3.9
```
Comment (zzstoatzz) [inline devops/infrastructure-as-code/azure/prefect-agent-on-aks/prefect.yaml:19]:
```suggestion
          image: prefecthq/prefect:2-python3.9
```

File: devops/infrastructure-as-code/azure/prefect-agent-on-aks/prefect.yaml [hunk 2]
```diff
@@ -14,20 +15,30 @@ spec:
...
+          command:
+            - prefect
+            - orion
```
Comment (zzstoatzz) [inline devops/infrastructure-as-code/azure/prefect-agent-on-aks/prefect.yaml:22]:
```suggestion
            - server
```

File: devops/infrastructure-as-code/azure/prefect-agent-on-aks/prefect.yaml [hunk 3]
```diff
@@ -14,20 +15,30 @@ spec:
         app: orion
```
Comment (zzstoatzz) [inline devops/infrastructure-as-code/azure/prefect-agent-on-aks/prefect.yaml:15]:
```suggestion
        app: server
```

File: devops/infrastructure-as-code/azure/prefect-agent-on-aks/prefect.yaml [hunk 4]
```diff
@@ -14,20 +15,30 @@ spec:
...
+            - start
+            - -host
+            - 0.0.0.0
+            - --log-level
+            - WARNING
+          imagePullPolicy: IfNotPresent
+          ports:
+            - containerPort: 4200
+        - name: agent
+          image: prefecthq/prefect:2.0b9-python3.9
```
Comment (zzstoatzz) [inline devops/infrastructure-as-code/azure/prefect-agent-on-aks/prefect.yaml:32]:
```suggestion
          image: prefecthq/prefect:2-python3.9
```

**Review by zzstoatzz — CHANGES_REQUESTED** [review]

**Comment by zzstoatzz** [comment]
@jamiezieziula conflict in `.github/workflows/python.yml` it appears

**Review by zzstoatzz — APPROVED** [review]

File: devops/github-actions/docker-build-push-gcp-artifact-registry.yaml [hunk 5]
```diff
@@ -1,49 +1,49 @@
...
       # Change the path `./Dockerfile` as needed!
       - name: Build the Docker image
         run: |
           docker build . --tag ${{ env.FLOW_IMAGE }} -f ./Dockerfile
 
-      # Generate a temporary access token from your service account's credentials
-      - id: 'auth'
-        name: 'Authenticate to Google Cloud'
-        uses: 'google-github-actions/auth@v0.6.0'
+      # Generate a temporary access token from your service accounts credentials
```
Comment (gabcoyne) [inline devops/github-actions/docker-build-push-gcp-artifact-registry.yaml:26]: This shouldn't have been changed

File: .github/workflows/pr-reminder.yaml [hunk 6]
```diff
@@ -1,6 +1,7 @@
+---
 name: PR Review Reminder
 
-on:
+"on":
```
Comment (gabcoyne) [inline .github/workflows/pr-reminder.yaml:4]: Why is this quoted?

**Review by gabcoyne — CHANGES_REQUESTED** [review]

File: devops/github-actions/docker-build-push-gcp-artifact-registry.yaml (hunk 5 above)
Comment (jamiezieziula) [reply devops/github-actions/docker-build-push-gcp-artifact-registry.yaml:26]: oop you're right

File: .github/workflows/pr-reminder.yaml (hunk 6 above)
Comment (jamiezieziula) [reply .github/workflows/pr-reminder.yaml:4]: on is considered a `bool` in yaml and so if it's not quoted - yaml linters yell

**Pull request closed without merging**